### DISCUSSION TITLE
Deli remixed

### DISCUSSION THREAD

**Discussion by Jigoku** [redeclipse/base#1558] [Maps and Content Showcase]
**Work in Progress**

Originally by LuckyStrike-Rx, here's a modern version ported to 2.1 now featuring an actual Deli staffed by robots;
Map layout is virtually identical other than 1 road tunnel was changed to a slope / obstacle. 

Known issues;
* dumpster model needs fixing
* lamp model needs fixing
* sign texture needed for bar/pub thing

* Upper middle area needs some geometry/texture fixing

[deli.zip](https://github.com/user-attachments/files/15844861/deli.zip)
![20240615_005105](https://github.com/redeclipse/base/assets/1535179/2aa248a3-dacd-4d29-a83a-4f079773f2cb)
![20240615_005240](https://github.com/redeclipse/base/assets/1535179/9591da3c-aaa5-45bc-bf0b-c5e9638ff8bf)
![20240615_005217](https://github.com/redeclipse/base/assets/1535179/42fc9d5b-98b3-42d8-9e78-83f276dabae7)
![20240615_005111](https://github.com/redeclipse/base/assets/1535179/989e48ef-3e96-40ae-b119-efc38c9ef32b)

**Comment by MirceaKitsune** [comment]
Sorry for taking so long to get to this, just gave the map a try. It looks incredible and plays just as well! I love the amount of personality you gave it, definitely a must have for the 2.1.0 release. Noticed a few things I'd change, hope it's not too much and you can take a look at all points.

I couldn't find any major issues per say. Only actual bug would be with the dumpster model which clips the wall in a couple of places, just move it a bit in front or to the side and that should fix it.

![20240725_231323 0001](https://github.com/user-attachments/assets/6a96ad49-4bb7-4568-b5fe-37d71d4c4b66)
![20240725_231344](https://github.com/user-attachments/assets/0217f1f1-94af-4aa9-9822-8a11da31eff1)

Only parts I'm unsure of are some remaining low detail brushes which can probably be replaced with models that look better. Namely the rails on the interior of the outer buildings, as well as the ladder that leads to the upper part of the roof: They feel a bit too blocky and simple for a map made at this point in time.

![20240725_231919](https://github.com/user-attachments/assets/8027ca1e-3fc2-4ae8-aa53-2be8ca7a2bd9)
![20240725_232030](https://github.com/user-attachments/assets/f9975b08-2efd-455d-bf40-bb74fb7bbf2d)

I also think those crates should be replaced with the more detailed crate model rather than a textured box.

![20240725_232047](https://github.com/user-attachments/assets/cdd392a5-ad4d-47c1-8790-3e2b12c7d1e4)

Same goes for the water tower: There's a model of a tank that can be scaled and used to look even better than this.

![20240725_232935](https://github.com/user-attachments/assets/f3356e86-718d-4915-b01d-27a8b299884b)

In terms of texturing, only part that looks a little meh is this panel on the ceiling in the center building: Maybe I'm looking too much into it but it just looks rough and lacking detail, if you can change it with something better aligned that would be awesome.

![20240725_232917](https://github.com/user-attachments/assets/46148cfd-fd92-43d3-b0b8-0868ccb94261)

Something else I noticed after a second test: What's with the sky brushes randomly appearing around some pillars? If they're meant to help with parkour I verified you can't even run up them. I'm surprised they aren't causing the skybox to show through.

![20240725_234120](https://github.com/user-attachments/assets/ad0b7ca8-7172-479b-b0a7-ade590636907)
![20240725_234140](https://github.com/user-attachments/assets/f9b42469-b1c8-48db-874f-dd73208c71b6)

There's also an issue with the grass, it appears in an octagonal pattern when seen from above. This might be an engine bug with surface particles, not sure if it's a bad configuration in the map itself. Was Echo doing this too?

![20240725_234310](https://github.com/user-attachments/assets/44a2d836-a1b5-49c3-87bc-7e5a54f8e4a2)

Beyond that I'd make better use of the Normal and Alternate modes: Perhaps make one version an orange sunset, the other a darker night time. It gives things a lot of extra variation to have them as different as possible and giving off different moods.

The map is also a bit too silent except for the store in the middle; I'd add a bit more ambience without making it too loud. Speaking of audio I don't hear any reverb, I'd use audio zones in the tunnels and building interiors.

Lastly remember to cull textures and hit F8 after saving to recompile the lighting: I noticed a few round surfaces where the lighting was flat and recompiling fixed it, the engine doesn't seem to do this internally when saving a map.

**Reply by Calinou** [reply]
> There's also an issue with the grass, it appears in an octagonal pattern when seen from above. This might be an engine bug with surface particles, not sure if it's a bad configuration in the map itself. Was Echo doing this too?

It's not a bug; grass was always designed like this (also in Sauerbraten since Trooper Edition). It's not visible in most situations as you rarely look straight down onto grass from above.

One way to make this less visible would be to partially tilt grass when above/below it (kind of like billboarding, but clamped to a certain extent so that blass grades don't end up floating when looking straight down/up).

**Reply by MirceaKitsune** [reply]
That's good to know, thanks for confirming. Echo has grass too and I didn't see it there, so I wondered if maybe something broke within the map. Not a major issue since you'll only see the pattern if you look from above.

**Reply by Jigoku** [reply]
Noted 👍 

The smaller skycubes are more or less for helping to not get stuck against some edges, it can help to push the player away from annoying small corners etc. If you impulse towards the archway one you screenshotted it should "bounce" you to the side more than abruptly getting stuck like velcro. If you look behind the cubes there's some faces pushed back, that stops the culling in the engine.

**Reply by MirceaKitsune** [reply]
Cheers! Figured their purpose was something among those lines. I use the clip material for that purpose... of course those can't be diagonal, I imagine an invisible sky brush was used to get the slopes.

![20240725_235349](https://github.com/user-attachments/assets/e9661ca1-bf6d-44c0-904e-eedb4949e67a)

I also noticed waypoints were hit with the waypoint explosion issue. Which I never figured out if it's even a bug or a feature, but I ran into it on Cargo and couldn't waypoint everything because of it. Don't know if redoing them from scratch to fix this is worth it: To me at least that looks broken and I'm not sure if it may have a performance impact or confuse the navigation system to have so many unnecessary waypoints scattered in the air. Wish the engine had a command to automatically remove those on affected maps, if it works and doesn't needlessly waste CPU I guess we can live with it.
